app/services/forest_liana/has_many_getter.rb

```diff
@@ -23,7 +23,7 @@ def perform
     end
 
     def count
-      @records_count = @records.count
+      @records_count = @search_query_builder.perform(@records).count
```

Comment:
Looks cool, shouldn't we do this in `query_for_batch` and `records` too?
If so, maybe the `@search_query_builder.perform` should be done `prepare_query`?

Reply:
Nice catch! Fixed!